google/ggrc-core
# GGRC-6367: Flask login spoofing

# Issue description

The login session is inconsistent after database redeployment
When the database gets updated while users perform the activity,
ids in the sessions and in the database may mess up.
This means that a random user can access the data of another.

# Steps to test the changes

An integration test was added to verify the fix.

Manually:

Log in and while keeping the session alive, update the database so that your user has a different id in the database. Your following requests should log you out

# Solution description

Fix login session inconsistency after database redeployment

For the sake of security, the user_email_hash is added to the session.
The session now has the user_email_hash
generated by generate_hash. This hash should be used
to confirm that a user with provided id has the same email stored
in the database

# Sanity checklist




# PR Review checklist
